LoopKit/Meal.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  Meal.swift
+//  LoopKit
+//
+//  Created by Noah Brauner on 7/13/23.
+//  Copyright © 2023 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+
+public struct Meal: Identifiable, Equatable {
+    public var id = UUID().uuidString
+    
+    public var carbsQuantity: HKQuantity
+    public var foodType: String
+    public var absorptionTime: TimeInterval
+    
+    public var name: String
+    
+    public init(carbsQuantity: HKQuantity, foodType: String, absorptionTime: TimeInterval, name: String) {
+        self.carbsQuantity = carbsQuantity
+        self.foodType = foodType
+        self.absorptionTime = absorptionTime
+        self.name = name
+    }
+    
+    public var carbsString: String {
+        let quantityFormatter = QuantityFormatter(for: .gram())
+        guard let string = quantityFormatter.string(from: carbsQuantity) else {
+            assertionFailure("Unable to format \(String(describing: carbsQuantity)) into gram format")
+            return ""
+        }
+        return string
+    }
+    
+    public var absorptionTimeString: String {
+        let absorptionFormatter: DateComponentsFormatter = {
+            let formatter = DateComponentsFormatter()
+            formatter.collapsesLargestUnit = true
+            formatter.unitsStyle = .full
+            formatter.allowsFractionalUnits = true
+            formatter.allowedUnits = [.hour, .minute]
+            return formatter
+        }()
```

Comment:
I might suggest making this `absorptionFormatter` a property of `Meal` directly. This is a computed property so this date formatter will get re-created every time it's called which is kind-of a heavy operation.

Reply:
Good catch! Thanks Cameron!